src/prof.c

```diff
@@ -2513,6 +2515,13 @@ prof_log_dummy_set(bool new_value) {
 }
 #endif
 
+/* Used as an atexit function to stop logging on exit. */
+static void
+prof_log_stop_final(void) {
+	tsd_t *tsd = tsd_fetch();
+	prof_log_stop(tsd_tsdn(tsd));
+}
+
 bool
 prof_log_start(tsdn_t *tsdn, const char *filename) {
 	if (!opt_prof || !prof_booted) {
```

Comment:
`prof_booted` is set after prof boot finishes. `prof_log_start` at initializing looks always invalid...